waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -191,6 +185,14 @@ proc new*(T: type PeerManager,
           backoffFactor = BackoffFactor,
           maxFailedAttempts = MaxFailedAttempts,): PeerManager =
 
+  let capacity = switch.peerStore.capacity
+  let maxConnections = switch.connManager.inSema.size
+  if maxConnections > capacity:
+    error "Max number of connections can't be greater than PeerManager capacity",
+         capacity = capacity,
+         maxConnections = maxConnections
+    doAssert(false, "Max number of connections can't be greater than PeerManager capacity")
```

Comment:
Raising a `Defect` is the preferred way to finish a program in Nim. It is the `panic` equivalent of Go or Rust.

Reply:
sure, i think i saw doAssert somewhere in libp2p and nimbus, but switched to raising Defect here.
3df173c6b77edf7bcfd049ed197dffc675bccede

added also a test case, since its important.